### PR TITLE
Proper handling of max-age and expires for cookies

### LINKED ISSUE #5033: [rfc] Making HTTP::Cookie fully? comply with RFC 6265.
Edited for correctness.
It is possible the below isn't needed, if we aren't trying to support Internet Explorer as a client to a Crystal server.

Looking at HTTP::cookie (in the latest Crystal revision), we aren't complying with RFC 6265 as regards `max-age` and `expires`.
See:
https://tools.ietf.org/html/rfc6265#section-4.1.2.2

https://tools.ietf.org/html/rfc6265#section-5.2.2

1. According to these, we should be noting a cookie as expired when the max-age header <=0 (assuming we aren't attempting to emulate Internet Explorer). This _is _done currently, but it discards the expires header.
2. We should be storing max-age as a separate value, and should be able to determine it's actual time based on the creation time of the cookie.
3. The `max_age` and `expires` values should both be modifiable so that exact cookies can be created as desired.

I've got a PR I'm working on to do the following:
1. add creation_time to each cookie, set to when #initialize is run.
2. change #expires to return the `expires` header value.
3. make max_age a Time::Span
4. generate string cookies using max-age and expires if they are defined.

Issues:
1. #expires now returns the `expires` value, not the actual expiration time (which can be dependent on max-age if provided). I don't have a suitable name to replace this method with. In other words, I need a method name for a method to give the calculated expiration time for a cookie.

Thoughts would be appreciated.

### PR DESCRIPTION
**Pull request by straight-shoota**
Reissue of #7925. I can't add commits to the original PR, so this is a fresh and more focused start.

This patch adds `max_age` and `creation_time` properties to `HTTP::Cookie` which allow a more versatile handling of expiration times.

Resolves #5033
Closes #7925

### CODE REVIEW

**Review by sdogruyol — APPROVED** [review]
Thank you @straight-shoota 🙏

File: spec/std/http/cookie_spec.cr [hunk 1]
```diff
@@ -297,42 +297,75 @@ module HTTP
...
+      end
+
+      it "sets expiration_time with old date" do
+        cookie = parse_set_cookie("bla=1; expires=Thu, 01 Jan 1970 00:00:00 -0000")
+        cookie.expiration_time.should eq Time.utc(1970, 1, 1, 0, 0, 0)
+      end
+
+      it "sets future expiration_time with max-age" do
+        cookie = parse_set_cookie("bla=1; max-age=1")
+        (cookie.expiration_time.not_nil!).should be > Time.utc
```
Comment (beta-ziliani) [inline spec/std/http/cookie_spec.cr:323]: Are these parentheses necessary?

**Review by beta-ziliani — APPROVED** [review]
just with a minor comment

File: spec/std/http/cookie_spec.cr (hunk 1 above)
Comment (straight-shoota) [reply spec/std/http/cookie_spec.cr:323]: No, I'll remove them.

**Pull request merged**